subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -600,6 +599,9 @@ private boolean tryAcquireLocksForNode(Node node, WorkerLeaseRegistry.WorkerLeas
         } else if (doesDestroyNotYetConsumedOutputOfAnotherNode(node, mutations.destroyablePaths)) {
             return false;
         }
+        if (mutations.hasValidationProblem) {
+            invalidNodeRunning = true;
```

Comment:
I don't think this belongs in `tryAcquireLocksForNode`. Moreover, it is not clear to me that we enter `finishedExecuting()`, where we reset the flag, when we skip the node since some dependencies failed. I think we only should set the flag when we are starting to execute a node.

Reply:
Moreover, it seems like there should be a test for that case: A task with a failed dependency which fails validation shouldn't block executing the remaining tasks.

Reply:
We do have two tests, one for not starting an invalid work when any other work is running, and another to test that we don't start any work when invalid work is running. What else do you think we need?

I think the test we have is sufficient, and adding an integration test would be too brittle anyway.

Reply:
I moved the flag further.

Reply:
> We do have two tests

We do not have a single integration test (aka E2E test). We have two unit tests, which I don't think is sufficient.